torchao/dtypes/nf4tensor.py

```diff
@@ -1,20 +1,25 @@
 import functools
 from dataclasses import dataclass
 from typing import Dict, Tuple
+import math
+import sys
 
 import torch
 import torch.nn.functional as F
 from torch import Tensor
+from torch.distributed.device_mesh import DeviceMesh
+from torch._prims_common import make_contiguous_strides_for
 
 
 aten = torch.ops.aten
 
 c10d_functional = torch.ops.c10d_functional
 
-from typing import Any, Tuple
+from typing import Any, Optional, Tuple, Union, List
 
 NF4_OPS_TABLE: Dict[Any, Any] = {}
 
+INNER_TENSOR_NAMES_FOR_FSDP = ["quantized_scalers", "quantization_factor", "quantized_data"]
```

Comment:
I exclude two tiny tensors: nf4 (numel=16) and scaler_mean (numel=1)
when GPU > numel, we need to implement padding for inner tensors. it's not worth the time in my opinion

Reply:
This seems like it'd apply to more than just FSDP. Is that correct?

Reply:
> This seems like it'd apply to more than just FSDP. Is that correct?

it applies general distributed case when we shard a single tensor to N GPUs. I can change the name to `INNER_TENSOR_NAMES_FOR_SHARDING` if that's clearer